crates/holochain/src/conductor/conductor.rs

```diff
@@ -1336,28 +1336,24 @@ mod network_impls {
 mod app_impls {
     use super::*;
 
+    use crate::sweettest::DnaWithRole;
+
     impl Conductor {
         #[cfg(feature = "test_utils")]
-        // FIXME: can rewrite now in terms of `get_install_app_payload_from_dnas`
         pub(crate) async fn install_app_legacy(
             self: Arc<Self>,
             installed_app_id: InstalledAppId,
-            cell_data: Vec<(InstalledCell, Option<MembraneProof>)>,
+            agent_key: AgentPubKey,
+            data: &[(impl DnaWithRole, Option<MembraneProof>)],
         ) -> ConductorResult<()> {
-            crate::conductor::conductor::genesis_cells(
-                self.clone(),
-                cell_data
-                    .iter()
-                    .map(|(c, p)| (c.as_id().clone(), p.clone()))
-                    .collect(),
+            let payload = crate::sweettest::get_install_app_payload_from_dnas(
+                installed_app_id,
+                agent_key,
+                data,
             )
-            .await?;
-
-            let cell_data = cell_data.into_iter().map(|(c, _)| c);
-            let app = InstalledAppCommon::new_legacy(installed_app_id, cell_data)?;
+            .await;
 
-            // Update the db
-            let _ = self.add_disabled_app_to_db(app).await?;
+            self.install_app_bundle(payload).await?;
 
             Ok(())
         }
```

Comment:
This is the main thrust of this PR: I wanted to not call `genesis_cells` in multiple places, since I'm about to add some DPKI-specific nuance to that function

Reply:
So this is one step towards deprecating `install_app_legacy`? Do you want to decorate that fn already as such?

Reply:
It's still useful, it's basically a way of providing minimal info to create a valid AppBundle and using that. Perhaps we could rename it. `install_app_minimal`? `_basic`?